websites/management/commands/markdown_cleaning/baseurl_rule.py

```diff
@@ -67,19 +61,28 @@ def get_content_by_url(self, website_id, content_relative_url: str):
             return self.website_contents[(website_id, dirpath, filename)]
 
 
-class BaseurlReplacer:
-    """Replacer function for use with WebsiteContentMarkdownCleaner. Replaces
+def get_all_website_content():
+    return WebsiteContent.all_objects.all().only(
+        "dirpath", "filename", "markdown", "website_id"
+    )
+
+
+class BaseurlReplacementRule(MarkdownCleanupRule):
+    """Replacement rule for use with WebsiteContentMarkdownCleaner. Replaces
     baseurl links with < resource_link > shortcodes.
 
     This is intentially limited in scope for now. Some baseurl links, such as
     those whose titles are images or include square brackets, are excluded from
     replacement.
     """
 
-    baseurl_regex = r"\[(?P<title>[^\[\]\n]*?)\]\({{< baseurl >}}(?P<url>.*?)\)"
+    regex = r"\[(?P<title>[^\[\]\n]*?)\]\({{< baseurl >}}(?P<url>.*?)\)"
+
+    alias = "baseurl"
 
-    def __init__(self, content_lookup: ContentLookup):
-        self.content_lookup = content_lookup
+    def __init__(self):
+        website_contents = get_all_website_content()
+        self.content_lookup = ContentLookup(website_contents)
```

Comment:
Previously the django command was passing `website_contents` to this replacer function. 

Now:
1. The django command gets all `website_content` so that it can iterate over them, making replacements
2. separately, this replacer function also gets all `website_contents` so that it can create a lookup to find content by url.

So there's some redundant fetching from the database... but it seemed simpler to make the rule responsible for getting its own data if it needs it.